src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -1404,6 +1152,7 @@ public void combineAllCommitsIntoInitialCommit(Repository repo) throws IllegalSt
      * @throws IOException if the deletion of the repository failed.
      */
     public void deleteLocalRepository(Repository repository) throws IOException {
+        super.deleteLocalRepository(repository);
```

Comment:
We can remove the rest of the method and only keep `cachedRepositories.remove(repoPath);`